docs/specs/build/manifest.yml

```diff
@@ -14,16 +14,172 @@ outputs:
+      ]
+    }
+---
+# `build.manifest` contains Toc dependencies
+inputs:
+  docfx.yml:
+  docs/a.md:
+  docs/TOC.md: |
+    # [file reference](a.md)
+outputs:
+  docs/a.json:
+  docs/TOC.json:
+  build.manifest: |
+    {
+      "dependencies":
+      [
+        {
+          "source": "docs/TOC.md",
+          "dependencies": 
+          [
+            {
+              "source": "docs/a.md",
+              "type": "tocLink"
+            }
+          ]
+        }
+      ]
+    }
+---
+# `build.manifest` contains Inclusion dependencies
+inputs:
+  docfx.yml:
+  docs/a.md: a [!INCLUDE[](b.md)]
+  docs/b.md:
+outputs:
+  docs/a.json:
+  docs/b.json:
+  build.manifest: |
+    {
+      "dependencies":
+      [
+        {
+          "source": "docs/a.md",
+          "dependencies": 
+          [
+            {
+              "source": "docs/b.md",
+              "type": "inclusion"
+            }
+          ]
+        }
+      ]
+    }
+---
+# `build.manifest` contains File(Link) dependencies
+inputs:
+  docfx.yml:
+  docs/a.md: Link to [b](b.md)
+  docs/b.md:
+outputs:
+  docs/a.json:
+  docs/b.json:
+  build.manifest: |
+    {
+      "dependencies":
+      [
+        {
+          "source": "docs/a.md",
+          "dependencies": 
+          [
+            {
+              "source": "docs/b.md",
+              "type": "link"
+            }
+          ]
+        }
+      ]
+    }
+---
+# `build.manifest` contains nested(token/codesnippet) dependencies
+inputs:
+  docfx.yml: |
+    content: docs/a.md
+  docs/a.md: a [!INCLUDE[](b.md)
+  docs/b.md: |
+    Link to [c](c.md)
+    b [!INCLUDE[](c.md)
+  docs/c.md:
+outputs:
+  docs/a.json:
+  build.manifest: |
+    {
+      "dependencies":
+      [
+        {
+          "source": "docs/a.md",
+          "dependencies": 
+          [
+            {
+              "source": "docs/b.md",
+              "type": "inclusion"
+            }
+          ]
+        },
+        {
+          "source": "docs/b.md",
+          "dependencies": 
+          [
+            {
+              "source": "docs/c.md",
+              "type": "link"
+            },
+            {
+              "source": "docs/c.md",
+              "type": "inclusion"
+            }
+          ]
+        }
+      ]
+    }
+---
+# `build.manifest` contains nested(toc) dependencies
+inputs:
+  docfx.yml:
+  docs/TOC.md: |
+    # [a](a/TOC.md)
+  docs/a/TOC.md: |
+    # [b](b.md)
+  docs/a/b.md:
+outputs:
+  docs/TOC.json:
+  docs/a/b.json:
+  build.manifest: |
+    {
+      "dependencies":
+      [
+        {
+          "source": "docs/a/TOC.md",
+          "dependencies": 
+          [
+            {
+              "source": "docs/a/b.md",
+              "type": "tocLink"
+            }
+          ]
+        },
+        {
+          "source": "docs/TOC.md",
+          "dependencies": 
+          [
+            {
+              "source": "docs/a/TOC.md",
+              "type": "tocInclusion"
+            }
+          ]
         }
       ]
     }
```

Comment:
> } [](start = 3, length = 2)

suggest adding a circular reference case

Reply:
circular reference cases are covered by TOC/Markdown spec test

---
In reply to: [193960655](https://github.com/dotnet/docfx/pull/2859#discussion_r193960655) [](ancestors = 193960655)